webapp/src/containers/LinkModal/index.tsx

```diff
@@ -1,138 +1,126 @@
+import plugin_constants from 'plugin_constants';
 
 const LinkModal = () => {
-    const [state, setState] = useState({
-        linkOrganization: '',
-        linkProject: '',
+    // State variables
+    const [projectDetails, setProjectDetails] = useState<LinkPayload>({
+        organization: '',
+        project: '',
     });
-    const [error, setError] = useState({
-        linkOrganizationError: '',
-        linkProjectError: '',
+
+    const [errorState, setErrorState] = useState<LinkPayload>({
+        organization: '',
+        project: '',
     });
-    const [linkPayload, setLinkPayload] = useState<LinkPayload | null>();
+
+    // Hooks
     const usePlugin = usePluginApi();
-    const {visibility, organization, project} = usePlugin.state['plugins-mattermost-plugin-azure-devops'].openLinkModalReducer;
-    const [loading, setLoading] = useState(false);
-    const [APIError, setAPIError] = useState('');
     const dispatch = useDispatch();
 
-    useEffect(() => {
-        if (organization && project) {
-            setState({
-                linkOrganization: organization,
-                linkProject: project,
-            });
-        }
-    }, [visibility]);
-
     // Function to hide the modal and reset all the states.
-    const onHide = useCallback(() => {
-        setState({
-            linkOrganization: '',
-            linkProject: '',
+    const resetModalState = () => {
+        setProjectDetails({
+            organization: '',
+            project: '',
         });
-        setError({
-            linkOrganizationError: '',
-            linkProjectError: '',
+        setErrorState({
+            organization: '',
+            project: '',
         });
-        setLinkPayload(null);
-        setLoading(false);
-        setAPIError('');
         dispatch(hideLinkModal());
-    }, []);
+    };
+
+    // Set organization name
+    const onOrganizationChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setErrorState({...errorState, organization: ''});
+        setProjectDetails({...projectDetails, organization: (e.target as HTMLInputElement).value});
+    };
 
-    const onOrganizationChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setError({...error, linkOrganizationError: ''});
-        setState({...state, linkOrganization: (e.target as HTMLInputElement).value});
-    }, [state, error]);
+    // Set project name
+    const onProjectChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setErrorState({...errorState, project: ''});
+        setProjectDetails({...projectDetails, project: (e.target as HTMLInputElement).value});
+    };
 
-    const onProjectChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setError({...error, linkProjectError: ''});
-        setState({...state, linkProject: (e.target as HTMLInputElement).value});
-    }, [state, error]);
+    // Handles on confirming link project
+    const onConfirm = () => {
```

Comment:
Use `useCallback`.

Reply:
I think we don't need this here as we are not caching anything here and it's called only 1 time on each user interaction only and no state variables are changing so we don't have any dependency to observe the change